controllers/consoleplugin/consoleplugin_reconciler.go

```diff
@@ -172,15 +194,20 @@ func pluginNeedsUpdate(plg *osv1alpha1.ConsolePlugin, desired *pluginSpec, ns st
 		plg.Spec.Service.Port != desired.Port
 }
 
-func deploymentNeedsUpdate(depl *appsv1.Deployment, desired *flowsv1alpha1.FlowCollectorSpec, ns string) bool {
+func deploymentNeedsUpdate(depl *appsv1.Deployment, desired *flowsv1alpha1.FlowCollectorSpec, ns string, cmDigest string) bool {
 	if depl.Namespace != ns {
 		return true
 	}
 	return containerNeedsUpdate(&depl.Spec.Template.Spec, &desired.ConsolePlugin) ||
 		hasLokiURLChanged(depl, &desired.Loki) ||
+		configChanged(&depl.Spec.Template, cmDigest) ||
```

Comment:
looks like we missed checking here the command line arguments .. so, changing the logLevel (or Loki URL) should trigger updating the deployment, which doesn't happen today.
As I'm testing it locally, I can do the changes and send a PR on top of yours